src/Microsoft.VisualStudio.Composition.Analyzers/VSMEF007DuplicateImportAnalyzer.cs

```diff
@@ -0,0 +1,167 @@
+        });
+    }
+
+    private static void AnalyzeType(SymbolAnalysisContext context)
+    {
+        var namedType = (INamedTypeSymbol)context.Symbol;
+
+        // Skip interfaces, enums, and other non-class types
+        if (namedType.TypeKind is not (TypeKind.Class or TypeKind.Struct))
+        {
+            return;
+        }
+
+        var importContracts = new List<ImportContract>();
+
+        // Collect all imports from properties
+        foreach (var member in namedType.GetMembers())
+        {
+            if (member is IPropertySymbol property)
+            {
+                var importAttribute = Utils.GetImportAttribute(property.GetAttributes());
+                if (importAttribute is not null)
+                {
+                    var contract = GetImportContract(importAttribute, property.Type);
+                    if (contract is not null)
+                    {
+                        importContracts.Add(new ImportContract(contract, property.Name, property.Locations[0]));
+                    }
+                }
+            }
+        }
+
+        // Collect all imports from constructor parameters
+        foreach (var member in namedType.GetMembers())
+        {
+            if (member is IMethodSymbol method && method.MethodKind == MethodKind.Constructor)
+            {
+                // Check if this constructor has [ImportingConstructor] or if it's the only constructor
+                bool isImportingConstructor = Utils.HasImportingConstructorAttribute(method) ||
+                    (namedType.Constructors.Length == 1 && Utils.HasInstanceExports(namedType));
+
+                if (isImportingConstructor)
+                {
+                    foreach (var parameter in method.Parameters)
+                    {
+                        var importAttribute = Utils.GetImportAttribute(parameter.GetAttributes());
+                        if (importAttribute is not null)
+                        {
+                            var contract = GetImportContract(importAttribute, parameter.Type);
+                            if (contract is not null)
+                            {
+                                importContracts.Add(new ImportContract(contract, parameter.Name, parameter.Locations[0]));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // Find duplicates
+        var contractGroups = importContracts.GroupBy(ic => ic.Contract).Where(g => g.Count() > 1);
+
+        foreach (var group in contractGroups)
+        {
+            foreach (var import in group)
+            {
+                context.ReportDiagnostic(Diagnostic.Create(
+                    Descriptor,
+                    import.Location,
+                    namedType.Name,
+                    import.Contract));
+            }
+        }
+    }
+
+    private static string? GetImportContract(AttributeData importAttribute, ITypeSymbol importType)
```

Comment:
It's actually far more complicated than this (see `ContractNameServices`), but since analyzers can't reference the library and I wouldn't want to reproduce all that logic here, a rough approximation is fine since it's not about what the contract name is _exactly_ as much as just identifying duplicates. But for clarity, that's probably worth commenting here.